File: Examples/VisualBasic/ModifyingAndConvertingImages/ApplyGaussWienerFilterForColoredImage.vb
Imports Aspose.Imaging.ImageFilters.FilterOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Class ApplyGaussWienerFilterForColoredImage
        Public Shared Sub Run()
            ' To get proper output please apply a valid Aspose.Imaging License. You can purchase full license or get 30 day temporary license from http:// Www.aspose.com/purchase/default.aspx.")
            ' ExStart:ApplyGaussWienerFilterForColoredImage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            ' Load the image
            Using image__1 As Image = Image.Load(dataDir & Convert.ToString("asposelogo.gif"))
                ' Caste the image into RasterImage
                Dim rasterImage As RasterImage = TryCast(image__1, RasterImage)
                If rasterImage Is Nothing Then
                    Return
                End If

                ' Create an instance of GaussWienerFilterOptions class and set the radius size and smooth value.
                Dim options As New GaussWienerFilterOptions(5, 1.5)
                options.Brightness = 1

                ' Apply MedianFilterOptions filter to RasterImage object.
                rasterImage.Filter(image__1.Bounds, options)

                ' Save the resultant image
                image__1.Save(dataDir & Convert.ToString("ApplyGaussWienerFilter_out.gif"))
            End Using
            ' ExEnd:ApplyGaussWienerFilterForColoredImage
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/ApplyGaussWienerFilterForColoredImage.cs
// GIST-ID: dbf48a70afb09037416f8aa3905e092a
using Aspose.Imaging;
using Aspose.Imaging.ImageFilters.FilterOptions;
using System;

/*
This project uses the automatic package restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    class ApplyGaussWienerFilterForColoredImage
    {
        public static void Run()
        {
            Console.WriteLine("Running example ApplyGaussWienerFilterForColoredImage");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            // Load the image
            using (Image image = Image.Load(dataDir + "asposelogo.gif"))
            {
                // Cast the image into RasterImage
                RasterImage rasterImage = image as RasterImage;
                if (rasterImage == null)
                {
                    return;
                }

                // Create an instance of GaussWienerFilterOptions class and set the radius size and smooth value.
                GaussWienerFilterOptions options = new GaussWienerFilterOptions(5, 1.5);
                options.Brightness = 1;

                // Apply GaussWienerFilterOptions filter to the RasterImage object and save the resultant image.
                rasterImage.Filter(image.Bounds, options);
                image.Save(dataDir + "ApplyGaussWienerFilter_out.gif");
            }

            Console.WriteLine("Finished example ApplyGaussWienerFilterForColoredImage");
        }
    }
}